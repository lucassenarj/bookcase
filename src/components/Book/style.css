.blog-details-area {
  background-color: #FFFFFF;
}

.page-title span {
	font-weight: 700;
	color: #f14836;
	display: contents
}

.details-image img {
	width: 30%;
  float: left;
  margin-right: 20px;
}

@media(max-width:767px) {
	.details-image img {
    width: 100%;
    float: none;
    margin-right: 0px;
  }
  
}

.blog-details-content .details-title {
	font-size: 24px;
	font-weight: 600;
	max-width: 680px;
	margin-top: 45px
}

@media(max-width:767px) {
	.blog-details-content .details-title {
		font-size: 20px
	}
}

.blog-details-content .text {
	margin-top: 20px
}

.blog-details-content .blog-share span {
	font-size: 14px;
	font-weight: 700;
	color: #222;
	font-family: nunito, sans-serif
}

.blog-details-content .blog-share .social {
  margin-top: 10px;
}

.blog-details-content .blog-share .social li {
	display: inline-block;
	margin-left: 10px;
	font-size: 16px;
	color: #a4a4a4;
	-webkit-transition: all .3s ease-out 0s;
	-moz-transition: all .3s ease-out 0s;
	-ms-transition: all .3s ease-out 0s;
	-o-transition: all .3s ease-out 0s;
	transition: all .3s ease-out 0s;
	margin-right: 20px;
	margin-top: 10px;
}

.blog-details-content .blog-share .social li i {
	margin-right: 8px
}

.blog-details-content .blog-share .social li:hover {
	color: #f14836
}

.blog-sidebar-wrapper {
	border: 1px solid #dedede4f
}

.blog-sidebar-wrapper .sidebar-title {
	padding: 15px 30px;
	border-bottom: 1px solid #dedede4f
}

.blog-sidebar-wrapper .sidebar-title .title {
	font-size: 18px;
	font-weight: 500
}

.blog-project .book-info {
	padding: 20px 30px 30px;
	border-bottom: 1px solid #dedede4f
}

.blog-project .book-info ul li {
	color: #798795;
	font-size: 14px;
	margin-top: 10px;
	-webkit-transition: all .3s ease-out 0s;
	-moz-transition: all .3s ease-out 0s;
	-ms-transition: all .3s ease-out 0s;
	-o-transition: all .3s ease-out 0s;
	transition: all .3s ease-out 0s;
	display: block;
}

.blog-project .book-info ul li span {
	float: right;
  font-weight: bold;
}

.blog-ratting .ratting-list {
	padding: 20px 30px 30px
}

.blog-ratting .ratting-list ul li {
	color: #798795;
	font-size: 14px;
	margin-top: 10px;
	-webkit-transition: all .3s ease-out 0s;
	-moz-transition: all .3s ease-out 0s;
	-ms-transition: all .3s ease-out 0s;
	-o-transition: all .3s ease-out 0s;
	transition: all .3s ease-out 0s;
	display: block
}

.blog-ratting .ratting-list ul li:hover {
	color: #f14836
}

.blog-ratting .ratting-list ul li span {
	float: right
}