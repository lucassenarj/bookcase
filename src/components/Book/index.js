import React from "react";
import PropTypes from "prop-types";
import i18n from "i18n";
import "./style.css";

function Book({ book }) {
  return (
    <section className="blog-details-area">
      <div className="container">
        <div className="row">
          <div className="col-lg-8">
            <div className="blog-details-content mt-50">
              <div className="details-image">
                <img src={book.imageLinks.thumbnail} alt={book.title} />
              </div>
              <h3 className="details-title">{ book.title }</h3>
              <div className="blog-share d-flex">
                <ul className="social">
                  <li><i className="fa fa-heart"></i> { i18n.t("components.book.like") }</li>
                  <li><i className="fa fa-share"></i> { i18n.t("components.book.share") }</li>
                  <li><i className="fa fa-list"></i> { i18n.t("components.book.save") }</li>
                </ul>
              </div>
              <p className="text">{ book.description }</p>
            </div>
          </div>

          <div className="col-lg-4">
            <div className="blog-sidebar mt-50">
              <div className="blog-sidebar-wrapper mb-60">
                <div className="blog-project">
                  <div className="sidebar-title text-center">
                    <h4 className="title">{ i18n.t("components.book.info") }</h4>
                  </div>
                  <div className="book-info">
                    <ul>
                      <li>
                      { i18n.t("components.book.author") }: <span>{ book.authors[0] }</span></li>
                      <li>{ i18n.t("components.book.publisher") }: <span>{ book.publisher }</span></li>
                      <li>{ i18n.t("components.book.pages") }: <span>{ book.pageCount }</span></li>
                      <li>{ i18n.t("components.book.date") }: <span>{ book.publishedDate }</span></li>
                      <li>{ i18n.t("components.book.language") }: <span> { book.language }</span></li>
                    </ul>
                  </div>
                </div>

                <div className="blog-project">
                  <div className="sidebar-title text-center">
                    <h4 className="title">{ i18n.t("components.book.identifiers") }</h4>
                  </div>
                  <div className="book-info">
                    <ul>
                      {
                        book.industryIdentifiers.map((identify, index) => (
                          <li key={index}>
                            {identify.type}
                            <span>{ identify.identifier }</span>
                          </li>
                        ))
                      }
                    </ul>
                  </div>
                </div>

                <div className="blog-ratting">
                  <div className="sidebar-title text-center">
                    <h4 className="title">{ i18n.t("components.book.my_books") }</h4>
                  </div>
                  <div className="ratting-list">
                    <ul>
                      <li>Like <span>50</span></li>
                      <li>Later <span>23</span></li>
                      <li>Finished <span>5</span></li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}

Book.propTypes = {
  book: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    imageLinks: PropTypes.object.isRequired,
    industryIdentifiers: PropTypes.arrayOf(PropTypes.object),
    authors: PropTypes.arrayOf(PropTypes.string).isRequired,
    pageCount: PropTypes.number.isRequired,
    publisher: PropTypes.string.isRequired,
    publishedDate: PropTypes.string,
    averageRating: PropTypes.number,
    language: PropTypes.string,
  }).isRequired,
}

export default Book;
