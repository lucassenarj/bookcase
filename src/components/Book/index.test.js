import React from "react";
import { render } from "@testing-library/react";
import Book from "./";
import bookItem from "./../../utils/mocks/book";


describe("Book Component", () => {
  it("renders correctly", () => {
    const { volumeInfo } = bookItem;
    const { container } = render(<Book book={volumeInfo} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});