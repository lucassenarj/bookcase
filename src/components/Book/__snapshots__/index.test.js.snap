// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Component renders correctly 1`] = `
<section
  class="blog-details-area"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-8"
      >
        <div
          class="blog-details-content mt-50"
        >
          <div
            class="details-image"
          >
            <img
              alt="Harry Potter and International Relations"
              src="http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=1&edge=curl&imgtk=AFLRE72B5UJgHNPjBlNhjuEpojz1haNfRzQSBz4lIbth_v3rkFm0YARlNEiDDxI-xSreA2QGDKU39PfbCeY8skvSa-IAimVoqHejyu9o31_V6N3wjHLVYxw&source=gbs_api"
            />
          </div>
          <h3
            class="details-title"
          >
            Harry Potter and International Relations
          </h3>
          <div
            class="blog-share d-flex"
          >
            <ul
              class="social"
            >
              <li>
                <i
                  class="fa fa-heart"
                />
                 
                components.book.like
              </li>
              <li>
                <i
                  class="fa fa-share"
                />
                 
                components.book.share
              </li>
              <li>
                <i
                  class="fa fa-list"
                />
                 
                components.book.save
              </li>
            </ul>
          </div>
          <p
            class="text"
          >
            Why not take seriously the claim that Harry Potter's world intertwines with our own? In this timely yet otherworldly volume, more than a dozen scholars of international relations join hands to demonstrate how this well-loved artifact of popular culture reflects and shapes our own lifeworld. A wide range of historical and sociological sources shows how Harry's world contains aspects of our own. Practices such as quidditch dovetail quite clearly with muggle sports, and the very British-ness of the books has, in translation into languages such as Turkish and Arabic, been transformed to reflect these unique cultures. Chapters on the political economy of the franchise as well as the scholarly problems of studying popular culture frame what is essentially a highly info-taining read. A landmark in muggle studies The Daily Prophet.
          </p>
        </div>
      </div>
      <div
        class="col-lg-4"
      >
        <div
          class="blog-sidebar mt-50"
        >
          <div
            class="blog-sidebar-wrapper mb-60"
          >
            <div
              class="blog-project"
            >
              <div
                class="sidebar-title text-center"
              >
                <h4
                  class="title"
                >
                  components.book.info
                </h4>
              </div>
              <div
                class="book-info"
              >
                <ul>
                  <li>
                    components.book.author
                    : 
                    <span>
                      Daniel H. Nexon
                    </span>
                  </li>
                  <li>
                    components.book.publisher
                    : 
                    <span>
                      Rowman & Littlefield
                    </span>
                  </li>
                  <li>
                    components.book.pages
                    : 
                    <span>
                      245
                    </span>
                  </li>
                  <li>
                    components.book.date
                    : 
                    <span>
                      2006
                    </span>
                  </li>
                  <li>
                    components.book.language
                    : 
                    <span>
                       
                      en
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="blog-project"
            >
              <div
                class="sidebar-title text-center"
              >
                <h4
                  class="title"
                >
                  components.book.identifiers
                </h4>
              </div>
              <div
                class="book-info"
              >
                <ul>
                  <li>
                    ISBN_10
                    <span>
                      0742539598
                    </span>
                  </li>
                  <li>
                    ISBN_13
                    <span>
                      9780742539594
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="blog-ratting"
            >
              <div
                class="sidebar-title text-center"
              >
                <h4
                  class="title"
                >
                  components.book.my_books
                </h4>
              </div>
              <div
                class="ratting-list"
              >
                <ul>
                  <li>
                    Like 
                    <span>
                      50
                    </span>
                  </li>
                  <li>
                    Later 
                    <span>
                      23
                    </span>
                  </li>
                  <li>
                    Finished 
                    <span>
                      5
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
